### PR TITLE
Explicit Dataset RDF and `training_data` in model RDF

### PR DESCRIPTION
**Pull request by FynnBe**
as discussed in #391 

Additionally this PR brings a readability update for the generated markdown docs. E.g. [the new model RDF docs](https://github.com/bioimage-io/spec-bioimage-io/blob/gh-pages-dataset_rdf/model_spec_latest.md)

### CODE REVIEW

File: bioimageio/spec/model/v0_4/raw_nodes.py [hunk 1]
```diff
@@ -117,9 +120,9 @@ class Model(_RDF):
     run_mode: Union[_Missing, RunMode] = missing
     sample_inputs: Union[_Missing, List[Union[URI, Path]]] = missing
     sample_outputs: Union[_Missing, List[Union[URI, Path]]] = missing
-    timestamp: datetime = missing
-    type: Literal["model"] = missing
-
     test_inputs: List[Union[URI, Path]] = missing
     test_outputs: List[Union[URI, Path]] = missing
+    timestamp: datetime = missing
+    training_data: Union[_Missing, Dict[str, Union[str, Dataset]]] = missing
```
Comment (constantinpape) [inline bioimageio/spec/model/v0_4/raw_nodes.py:126]: I think that we haven't fully settled on whether we want a `List` or a `Dict` here yet. Let's finalize that decision in the meeting tomorrow.

**Review by constantinpape — COMMENTED** [review]
I am still slightly in favor of having a list instead of a dict. I don't have a very strong opinion though, but we can maybe briefly discuss in the meeting tomorrow. Otherwise this looks good to me. cc @oeway

**Comment by constantinpape** [comment]
Also, the tests are still failing.

File: bioimageio/spec/model/v0_4/raw_nodes.py (hunk 1 above)
Comment (FynnBe) [reply bioimageio/spec/model/v0_4/raw_nodes.py:126]: as discussed in short meeting with @constantinpape @oeway : no list nor dict. single dataset id (or in-place RDF) and we'll take it from there when need be...

**Comment by FynnBe** [comment]
@oeway didn't you wonder about zenodo's occasional 503? https://github.com/bioimage-io/spec-bioimage-io/runs/5279715600?check_suite_focus=true

**Comment by oeway** [comment]
> @oeway didn't you wonder about zenodo's occasional 503? https://github.com/bioimage-io/spec-bioimage-io/runs/5279715600?check_suite_focus=true

Ok, but I was more wondering about the rate limit error (code 429 Too Many Requests), for 503, it means zenodo is out of service temporary which can happen and nothing we can do about.

**Pull request merged**